### DISCUSSION TITLE
Providing the paths to the pretrained models while running ocr_predictor()

### DISCUSSION THREAD

**Discussion by ds-anik** [mindee/doctr#1151] [Q&A]
Hi, 

I am kind of new to Using DocTR, but I have faced an issue and I woul appreciate if anyone can help me on this. I am trying to use the DocTR library on the dataiku platform (a data science studio) and I keep getting error while I am trying to run the ocr_predictor(). This happens because my user has not permission to download and store pretrained models in the cache of this platform. My question is that is there any way I can store the db_resnet50.pt and crnn_vgg16.pt in a local directory and provide the paths to the ocr_predictor function? is there anyway to use the pre-trained models without downloading directly to cache?

**Comment by TomekPro** [comment]
You can try sth like this:
```
    def __init__(self, detection_model_path, recognition_model_path) -> None:

        # Load detection model
        det_model = db_resnet50(pretrained=False)
        det_params = torch.load(detection_model_path, map_location="cpu")
        det_model.load_state_dict(det_params)

        # Load recognition model
        reco_model = crnn_vgg16_bn(pretrained=False, vocab=VOCABS["polish"])
        reco_params = torch.load(recognition_model_path, map_location="cpu")
        reco_model.load_state_dict(reco_params)

        # Instantiate predictor
        self.predictor = ocr_predictor(det_model, reco_model, pretrained=True)
```
